src/bootstrap/src/core/config/tests.rs

```diff
@@ -23,6 +23,19 @@ pub(crate) fn parse(config: &str) -> Config {
     )
 }
 
+fn get_toml(file: &Path) -> Result<TomlConfig, toml::de::Error> {
+    let contents = std::fs::read_to_string(file).unwrap();
+    toml::from_str(&contents).and_then(|table: toml::Value| TomlConfig::deserialize(table))
+}
+
+/// Helps with debugging by using consistent test-specific directories instead of
```

Comment:
The way I usually do it is to use temporary directories that don't cleanup themselves in case of panic. We should just add a dev dependency on `tempfile`, or at least on `rand` to have proper randomness rather than these hacks with clearing the directories.. But I don't want to block this, we can refactor it later.

Reply:
There are some `build_helpers` tests that should use `tempfile` too

Reply:
It's not a hack or something. I wanted to keep these test files under the temporary directory managed by bootstrap (which is under the build directory) so we can check them easily while running/debugging them. `tempfile` wouldn't be a good choice to achieve this.

Reply:
Sure, I understand that, but at least it would be nice to have a function for this to avoid having to parse an empty config to get a tempdir and then explicitly removing its contents at the start of each test :D But that's a small thing.

Reply:
Btw you can configure tempfile to create a unique random directory in a specific directory of your choice (https://docs.rs/tempfile/3.19.1/tempfile/struct.TempDir.html#method.new_in).

Reply:
> Btw you can configure tempfile to create a unique random directory in a specific directory of your choice (https://docs.rs/tempfile/3.19.1/tempfile/struct.TempDir.html#method.new_in).

That's nice, but it won't replace the current logic if you still want to keep these test files separated by test name for each individual test. It could be used to generate the file in the directory that the current logic finds but I'm not sure how good it would be to combine them instead of just sticking with one approach. 😄

Reply:
> Sure, I understand that, but at least it would be nice to have a function for this to avoid having to parse an empty config to get a tempdir and then explicitly removing its contents at the start of each test :D But that's a small thing.

That's easy work and I'm fine with doing it. But I'm not sure when I will have the time. I don't have access to my dev environment atm; I can only leave notes on Github atm. Anyone with time can handle it after this lands.

Reply:
Cleanup issue: #139959

Reply:
I think the issue doesn't share the same point with [this](https://github.com/rust-lang/rust/pull/138934#discussion_r2048448670).

Reply:
Oh oops I quoted the wrong comment I think

Reply:
I done that with the last commit along with the CI failure fix.